### PR TITLE
Quotes subscription

### PR DESCRIPTION
**Pull request by stalexxx**
allows to subcribe qoutes

### CODE REVIEW

File: src/data/v2/stream.rs [hunk 1]
```diff
@@ -260,6 +260,35 @@ pub struct Bar {
 }
 
 
+/// Aggregate data for an equity.
```
Comment (d-e-s-o) [inline src/data/v2/stream.rs:265]: Should this comment be adjusted?

File: src/data/v2/stream.rs [hunk 2]
```diff
@@ -260,6 +260,35 @@ pub struct Bar {
...
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+
+pub struct Quote {
+  /// The bar's symbol.
```
Comment (d-e-s-o) [inline src/data/v2/stream.rs:269]: Outdated comment.

File: src/data/v2/stream.rs [hunk 3]
```diff
@@ -260,6 +260,35 @@ pub struct Bar {
...
+
+  /// The ask's price.
+  #[serde(rename = "ap")]
+  pub ask_price: Num,
+  /// The ask's size.
+  #[serde(rename = "as")]
+  pub ask_size: u64,
+
+
+  /// The bar's time stamp.
```
Comment (d-e-s-o) [inline src/data/v2/stream.rs:292]: Can you please fix the comment?

File: src/data/v2/stream.rs [hunk 4]
```diff
@@ -281,6 +310,9 @@ pub enum DataMessage {
   /// A variant representing aggregate data for a given symbol.
   #[serde(rename = "b")]
   Bar(Bar),
+  /// A variant representing aggregate data for a given symbol.
```
Comment (d-e-s-o) [inline src/data/v2/stream.rs:319]: Adjust the comment?

File: src/data/v2/stream.rs [hunk 5]
```diff
@@ -326,13 +360,15 @@ impl subscribe::Message for ParsedMessage {
...
         DataMessage::Bar(bar) => subscribe::Classification::UserMessage(Ok(Ok(Data::Bar(bar)))),
+        DataMessage::Quotes(quote) => subscribe::Classification::UserMessage(Ok(Ok(Data::Quote(quote)))),
         DataMessage::Subscription(data) => {
           subscribe::Classification::ControlMessage(ControlMessage::Subscription(data))
         },
         DataMessage::Success => subscribe::Classification::ControlMessage(ControlMessage::Success),
         DataMessage::Error(error) => {
           subscribe::Classification::ControlMessage(ControlMessage::Error(error))
         },
+
```
Comment (d-e-s-o) [inline src/data/v2/stream.rs:377]: Please don't include spurious changes like this in a pull request.

File: src/data/v2/stream.rs [hunk 6]
```diff
@@ -401,6 +437,9 @@ impl<const N: usize> From<[&'static str; N]> for Normalized {
 pub struct MarketData {
   /// The aggregate bars to subscribe to.
   pub bars: Normalized,
+
+  /// The aggregate quotes to subscribe to.
```
Comment (d-e-s-o) [inline src/data/v2/stream.rs:464]: What is being aggregated with quotes? Aren't they "raw"?

File: src/data/v2/stream.rs [hunk 7]
```diff
@@ -412,6 +451,15 @@ impl MarketData {
   {
     self.bars = symbols.into();
   }
+
+  /// A convenience function for setting the [`bars`][MarketData::bars]
+  /// member.
```
Comment (d-e-s-o) [inline src/data/v2/stream.rs:479]: Please update.

File: src/data/v2/stream.rs [hunk 8]
```diff
@@ -44,6 +45,7 @@ use websocket_util::subscribe::MessageStream;
 use websocket_util::tungstenite::Error as WebSocketError;
 use websocket_util::wrap;
 use websocket_util::wrap::Wrapper;
+use crate::api::v2::asset::Exchange;
```
Comment (d-e-s-o) [inline src/data/v2/stream.rs:48]: Data and "API" are treated differently and we do not share publicly accessible data types between the two. Until Alpaca clarifies their relationship I don't intend to change that. I don't think I want to include the exchange here yet. Let's leave this change out for now.

**Review by d-e-s-o — CHANGES_REQUESTED** [review]

File: src/data/v2/stream.rs (hunk 5 above)
Comment (stalexxx) [reply src/data/v2/stream.rs:377]: sorry

File: src/data/v2/stream.rs (hunk 1 above)
Comment (stalexxx) [reply src/data/v2/stream.rs:264]: done

File: src/data/v2/stream.rs (hunk 2 above)
Comment (stalexxx) [reply src/data/v2/stream.rs]: done

File: src/data/v2/stream.rs (hunk 8 above)
Comment (stalexxx) [reply src/data/v2/stream.rs]: ok, changed that to just char

File: src/data/v2/stream.rs (hunk 7 above)
Comment (stalexxx) [reply src/data/v2/stream.rs]: fixed

File: src/data/v2/stream.rs (hunk 6 above)
Comment (stalexxx) [reply src/data/v2/stream.rs:443]: fixed

File: src/data/v2/stream.rs (hunk 4 above)
Comment (stalexxx) [reply src/data/v2/stream.rs]: fixed

File: src/data/v2/stream.rs (hunk 3 above)
Comment (stalexxx) [reply src/data/v2/stream.rs]: fixed

**Comment by stalexxx** [comment]
Fix all you mentioned

**Comment by stalexxx** [comment]
sorry for delay

File: src/data/v2/stream.rs (hunk 8 above)
Comment (d-e-s-o) [reply src/data/v2/stream.rs]: > ok, changed that to just char

Thanks! I am not sure a `char` is appropriate, to be honest. This crate is about providing a convenient interface to the upstream functionality. Let's leave this out for now. If you need it, I suggest we open a new issue and/or pull request and we can discuss. I am not terribly opposed to sharing data structure between `api` and `data`, but we may want to double check with Alpaca how independent they are.

**Comment by d-e-s-o** [comment]
Looks mostly good. I've made a few minor changes and merged the change. Thanks for your contribution!

**Pull request closed without merging**